### PR TITLE
Include sonar scan as github action

### PR DESCRIPTION
**Pull request by awilmore**
WIP

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# if you want disabled the DTD verification for a proxy problem for example, true by default
+sonar.coverage.dtdVerification=false
+
+# New config
+sonar.host.url=https://code.tools.mspenv.io/
+
+# Source settings
+sonar.sources=src
+
+# Coverage settings
```
Comment (mx51damon) [inline sonar-project.properties:13]: Hi @awilmore, not sure whether do we need this line or not `sonar.javascript.junit.reportPaths=./junit.xml`?

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+sonar.coverage.exclusions=**/*.test.ts
```
Comment (mx51damon) [inline sonar-project.properties:14]: And I got to mention that some of test file are `xxx.test.ts` and some of files are `xxx.test.tsx`. I thought we might need to include both?

Comment (awilmore) [reply sonar-project.properties:14]: Yep we can do that

File: sonar-project.properties (hunk 1 above)
Comment (awilmore) [reply sonar-project.properties:13]: I'm not sure either, but it seems to work without it. And from some of the examples I saw, the `sonar-project.properties` file only included:

```
sonar.javascript.lcov.reportPaths=coverage/lcov.info
```

which seems to be enough to collect coverage results.

Comment (mx51damon) [reply sonar-project.properties:13]: Yep sure, all good 🍻

**Review by mx51damon — APPROVED** [review]

**Comment by awilmore** [comment]
Hi @mx51damon , I'll keep this PR on hold until I finish working on the Github action for this, which I hope will be in a day or two.

**Comment by mx51damon** [comment]
Yep sure @awilmore, all good 🤗

**Comment by awilmore** [comment]
Closing this PR as I hit a road-block using Github Actions for this.

I like the idea of something updating Github PRs to be managed by Github Actions, but it won't work unfortunately. Github refuses to trigger a separate pipeline job when a **git tag is created**. I spent ages trying to figure out how to implement this whole thing using Github actions, but that failure to trigger on a tag means I need to look at implementing this using Azure Devops, unfortunately.

**Pull request closed without merging**